index.bs

```diff
@@ -18,5 +18,166 @@ Introduction here.
 Use cases {#usecases}
 =====================
 
-Use cases here.
+## High-Level Use Cases ## {#usecases-highlevel}
 
+This section illustrates application-level use cases for neural network
+inference hardware acceleration. All applications in those use cases can be
+built on top of pre-trained deep neural network (DNN) models.
+
+### Person Detection ### {#usecase-person-detection}
+
+A user is browsing a social media site and wishes to take a photo and upload it
+to the site. Before the photo is uploaded, the site does object detection (for
+example, using object detection approaches such as [[SSD]] or [[YOLO]] that use
+a single DNN) to detect regions that include persons so that the user can filter
+and de-personalize irrelevant persons on it.
+
+### Skeleton Detection ### {#usecase-skeleton-detection}
+
+The web application extracts her skeleton by running a machine learning model
+which allows for real-time human pose estimation such as [[PoseNet]] to
+recognize her gesture and body language. When she strikes a specified pose like
+raising a hand, the web application automatically takes a photo and upload it to
+an online storage.
+
+### Random Image Generation ### {#usecase-image-generation}
+
+A user wishes to make her new account and looks for a new icon image. When she
+clicks a "Generate" button on the webpage for creating an account, the webpage
+runs a generator model of generative adversarial network (GAN) for icon
+synthesis [[LogoSynthesis]]. She can repeat random icon
+generation until she finds her favorite one.
+
+## Low-Level Use Cases ## {#usecases-lowlevel}
+
+This section collects API-level use cases for a dedicated low-level API for
+neural network inference hardware acceleration. It is expected that Machine
+Learning frameworks will be key consumers of the Web Neural Network API (WebNN
+API) and the low-level details exposed through the WebNN API are abstracted out
+from typical web developers. However, it is also expected that web developers
+with specific interest and competence in Machine Learning will want to interface
+with the WebNN API directly instead of a higher-level ML framework.
+
+### Custom Layer ### {#usecase-custom-layer}
+
+A web application developer wants to run a DNN model on the WebNN API. However,
+she has found that some of activation functions like [[LeakyReLU]], [[ELU]],
+etc. are not included in the WebNN API. So she constructs custom layers of the
+additional activation functions on top of the WebNN API. Note that the scope of
+custom layers may include convolution, normalization, etc. as well as
+activation.
+
+### Network Concatenation ### {#usecase-network-concat}
```

Comment:
As mentioned in the TPAC F2F meeting, this looks like a training use case. As training is out of scope of current charter, would it be better to add this in the future?

Reply:
Correct, but not limited to training. Possible detailed examples are:

- The web app downloads convolutional layer weights of MobileNetV1/V2 from CDN and weights of fully-connected layers made by transfer learning from her own web site
- The web app downloads complete weights of MobileNetV1/V2, and then partially update fully-connected layers by downloading fine-tuned weights

Anyway, the current description seems to suggest the use case of training, as you pointed out. I'll update those sentences so that they clearly indicate a use case of client-side partial update based on fine tuning or transfer learning.